target/linux/mvebu/cortexa9/base-files/etc/init.d/fancontrol

```diff
@@ -0,0 +1,93 @@
+		fan1_pwm_path="/sys/class/hwmon/hwmon1/pwm1" #fan connected on j10 header on board
+		fan2_pwm_path="/sys/class/hwmon/hwmon2/pwm1" #fan connected on j17 header on board
+				
+		pwm_min="20" #lowest value for fan speed, with pwm lower than this value the fan is off
+		
+		cpu_temp_min="40000" #lowest value, if CPU is cooler than this fan is off
+		cpu_temp_max="85000" #max value, if CPU is hotter the fan is at max speed
+		;;
+	esac
+
+	#calculating 10 temperature steps
+	temp_step=$(( (cpu_temp_max - cpu_temp_min) / 10 ))
+	
+	target_temp_1=$(( cpu_temp_min + ( temp_step * 1 ) ))
+	target_temp_2=$(( cpu_temp_min + ( temp_step * 2 ) ))
+	target_temp_3=$(( cpu_temp_min + ( temp_step * 3 ) ))
+	target_temp_4=$(( cpu_temp_min + ( temp_step * 4 ) ))
+	target_temp_5=$(( cpu_temp_min + ( temp_step * 5 ) ))
+	target_temp_6=$(( cpu_temp_min + ( temp_step * 6 ) ))
+	target_temp_7=$(( cpu_temp_min + ( temp_step * 7 ) ))
+	target_temp_8=$(( cpu_temp_min + ( temp_step * 8 ) ))
+	target_temp_9=$(( cpu_temp_min + ( temp_step * 9 ) ))
+	target_temp_10=$cpu_temp_max
+	
+	#calculating pwm step
+	pwm_step=$(( ( 255 - pwm_min ) / 10 ))
+	
+	pwm_step_1=$(( pwm_min ))
+	pwm_step_2=$(( pwm_min + ( pwm_step * 1 ) ))
+	pwm_step_3=$(( pwm_min + ( pwm_step * 2 ) ))
+	pwm_step_4=$(( pwm_min + ( pwm_step * 3 ) ))
+	pwm_step_5=$(( pwm_min + ( pwm_step * 4 ) ))
+	pwm_step_6=$(( pwm_min + ( pwm_step * 5 ) ))
+	pwm_step_7=$(( pwm_min + ( pwm_step * 6 ) ))
+	pwm_step_8=$(( pwm_min + ( pwm_step * 7 ) ))
+	pwm_step_9=$(( pwm_min + ( pwm_step * 8 ) ))
+	pwm_step_10=$(( pwm_min + ( pwm_step * 9 ) ))
+	
+	
+	#running a constant loop to monitor temp and assign pwm every 10 seconds
+	while (true)
+	do
+	cpu_temp=$(cat $cpu_temp_path)
+		
+	if [ $cpu_temp -le $cpu_temp_min ]; then
+		pwm_value=0
+	elif [ $cpu_temp -gt $cpu_temp_min ] && [ $cpu_temp -le $target_temp_1  ]; then
+		pwm_value=$pwm_step_1
+	elif [ $cpu_temp -gt $target_temp_1 ] && [ $cpu_temp -le $target_temp_2 ]; then
+		pwm_value=$pwm_step_2
+	elif [ $cpu_temp -gt $target_temp_2 ] && [ $cpu_temp -le $target_temp_3 ]; then
+		pwm_value=$pwm_step_3
+	elif [ $cpu_temp -gt $target_temp_3 ] && [ $cpu_temp -le $target_temp_4 ]; then
+		pwm_value=$pwm_step_4
+	elif [ $cpu_temp -gt $target_temp_4 ] && [ $cpu_temp -le $target_temp_5 ]; then
+		pwm_value=$pwm_step_5
+	elif [ $cpu_temp -gt $target_temp_5 ] && [ $cpu_temp -le $target_temp_6 ]; then
+		pwm_value=$pwm_step_6
+	elif [ $cpu_temp -gt $target_temp_6 ] && [ $cpu_temp -le $target_temp_7 ]; then
+		pwm_value=$pwm_step_7
+	elif [ $cpu_temp -gt $target_temp_7 ] && [ $cpu_temp -le $target_temp_8 ]; then
+		pwm_value=$pwm_step_8
+	elif [ $cpu_temp -gt $target_temp_8 ] && [ $cpu_temp -le $target_temp_9 ]; then
+		pwm_value=$pwm_step_9
+	elif [ $cpu_temp -gt $target_temp_9 ] && [ $cpu_temp -le $target_temp_10 ]; then
+		pwm_value=$pwm_step_10
+	elif [ $cpu_temp -gt $target_temp_10 ]; then
+		pwm_value=255
+	fi
```

Comment:
Since elif will only enter one case, one should be able to reduce the conditions to one condition here for most cases:
```
	if [ $cpu_temp -le $cpu_temp_min ]; then
		pwm_value=0
	elif [ $cpu_temp -le $target_temp_1  ]; then
		pwm_value=$pwm_step_1
	elif [ $cpu_temp -le $target_temp_2 ]; then
		pwm_value=$pwm_step_2

...
```

Reply:
And the final elif would just be an "else" then ...